apps/aecore/lib/aecore/structures/data_tx.ex

```diff
@@ -72,25 +71,52 @@ defmodule Aecore.Structures.DataTx do
     accounts_state = chainstate.accounts
     tx_type_state = Map.get(chainstate, tx.type, %{})
 
-    {new_accounts_state, new_tx_type_state} =
-      tx.payload
-      |> tx.type.init()
-      |> tx.type.process_chainstate!(
-        tx.sender,
-        tx.fee,
-        tx.nonce,
-        accounts_state,
-        tx_type_state
-      )
-
-    new_chainstate =
-      if Map.has_key?(chainstate, tx.type) do
-        Map.put(chainstate, tx.type, new_tx_type_state)
-      else
+    case tx.payload
+         |> tx.type.init()
+         |> tx.type.process_chainstate!(
+           tx.sender,
+           tx.fee,
+           tx.nonce,
+           accounts_state,
+           tx_type_state
+         ) do
+      {:error, reason} ->
```

Comment:
Do not you think the more valuable information should be kept first place instead of `{: error, reason}` ?

Reply:
Yes, but in this case I can only pattern match the :error, since the value i need is `{new_accounts_state, new_tx_type_state}`